panel/custom.py

```diff
@@ -108,24 +108,45 @@ def __init__(self, **params):
         self._watching_esm = False
         self._event_callbacks = defaultdict(list)
 
-    @property
-    def _esm_path(self):
-        esm = self._esm
+    @classproperty
+    def _compiled_path(cls) -> os.PathLike | None:
+        if config.autoreload:
+            return None
+        try:
+            mod_path = pathlib.Path(inspect.getfile(cls)).parent
+        except (OSError, TypeError, ValueError):
+            return None
+        cls_name = camel_to_kebab(cls.__name__)
+        path = mod_path / f'{cls_name}.compiled.js'
+        return path if path.is_file() else None
+
+    @classmethod
+    def _esm_path(cls, compiled: bool = True) -> os.PathLike | None:
+        if compiled:
+            compiled_path = cls._compiled_path
+            if compiled_path:
+                return compiled_path
+        esm = cls._esm
         if isinstance(esm, pathlib.PurePath):
             return esm
         try:
-            esm_path = pathlib.Path(inspect.getfile(type(self))).parent / esm
+            if hasattr(cls, '__path__'):
+                mod_path = cls.__path__
```

Comment:
```suggestion
                mod_path = Path(cls.__path__[0])
```

![image](https://github.com/user-attachments/assets/53e32ad5-c420-4164-9d0b-c4bbc65de762)

Reply:
`__path__` is something I patch onto the classes in the dynamic module at compile time. Probably should have chosen some better name.